### PR TITLE
bootstrap: switch to bullseye, unpin docker

### PR DESCRIPTION
**Pull request by BenTheElder**
After this merges and builds we'll have to update kubekins to pickup the new image, then bump the CI images like https://github.com/kubernetes/test-infra/pull/28367

### CODE REVIEW

File: images/bootstrap/Dockerfile [hunk 1]
```diff
@@ -106,7 +106,7 @@ RUN curl -fsSL https://download.docker.com/linux/$(. /etc/os-release; echo "$ID"
 # Trying to remount these makes for a very noisy error block in the beginning of
 # the pod logs, so we just comment out the call to it... :shrug:
 RUN apt-get update && \
-    apt-get install -y --no-install-recommends docker-ce=5:20.10.* && \
```
Comment (BenTheElder) [inline images/bootstrap/Dockerfile:109]: Docker has been 20.10.X since 2020. This wasn't doing anything and docker has been plenty stable.

**Comment by BenTheElder** [comment]
cc @thockin

**Comment by BenTheElder** [comment]
KIND's similar image (images/krte) has been on bullseye for 8+ months now without issue.

**Comment by BenTheElder** [comment]
cc @aojea

**Comment by aojea** [comment]
/lgtm

🤞

**Comment by BenTheElder** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-test-infra-push-bootstrap/1612573806851788800

never so easy :-)

**Comment by aojea** [comment]
python2 deprecation, right?

**Comment by thockin** [comment]
At least one of our verify tools (boilerplate) calls a python2 program.
We'll have to fix that.

**Comment by BenTheElder** [comment]
Yes, py2, we fixed that elsewhere in the project before python2 was deprecated 😂 , of course k/k scripts are not yet 🙃

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/issues/114967

**Comment by BenTheElder** [comment]
now that https://github.com/kubernetes/kubernetes/issues/114967 appears to be fixed, time to resume patching this.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/pull/28465

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/pull/28466 to upgrade it in kubekins-e2e, then we can upgrade CI.

**Pull request merged**